neuralmonkey/decoders/decoder.py

```diff
@@ -30,19 +29,51 @@
     "LSTM": tf.contrib.rnn.LSTMCell
 }
 
-# pylint: disable=invalid-name
-RNNFeedables = extend_namedtuple(
-    "DecoderFeedables",
-    DecoderFeedables,
-    [("prev_rnn_state", tf.Tensor),
-     ("prev_rnn_output", tf.Tensor),
-     ("prev_contexts", List[tf.Tensor])])
-
-RNNHistories = extend_namedtuple(
-    "RNNHistories",
-    DecoderHistories,
-    [("attention_histories", List[Tuple])])  # AttentionLoopState and kids
-# pylint: enable=invalid-name
+
+# Inheriting namedtuples not yet fully supported in Python nor in mypy/pylint,
+# so we just copy the parent attributes.
+class RNNFeedables(NamedTuple(
```

Comment:
Moc se mi nelibi, ze tady ztracime to dedeni atributu od predka. Nejde to vymyslet nejak lip?

Reply:
Dědit namedtuply jako jinou třídu nejde. Může se to tam napsat dynamicky (tj třeba říct `NamedTuple("name", fields_of_parent  + [("a": int)])`). Jenže to pak mypy neumí kontrolovat typ. Jednak řekne že to je illegal parent a jednak řekne u jakýhokoliv referencování fieldů z parenta, že tam ten field neni.

Dalo by se to nahradit normálníma třídama, ale s těma by pak nefungoval ten while cyklus a ani to `nest.map_structure`.

Tohle mi zkrátka přijde nejlepší řešení. Navíc se ta dědičnost nepoužívá. Když děláš RNN feedables, tak si napřed uděláš DecoderFeedables a pak jima osadíš novej objekt RNN feedablů.